### ISSUE TITLE
space-in-parens: a space after parenthesis after keyword

### ISSUE THREAD

**Issue by roman4e** [eslint/eslint#10413]
**What rule do you want to change?**
Using space-in-parens you cannot split when space needed or not, it always rely on the same rule/behavior. For example, you can't warn about space in parentheses in function call but skip warning in parentheses in keyword if/while/switch usage. 

**Does this change cause the rule to produce more or fewer warnings?**
Might produce fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
As new option `afterKeyword`

**Please provide some example code that this change will affect:**

```js
if ( condition && condition2 ) 
{
...
someFunction(arg1, arg2, arg3); 
```

**What does the rule currently do for this code?**
It produces a warning/error in `if ( condition && condition2 )` when `space-in-parens` set to `"never"`

**What will the rule do after it's changed?**
Does not produce warning/error if `afterKeyword` set to `true` and `space-in-parens` set to `"never"`.

**Comment by Wulfex** [comment]
Agree with this 100%. Would like to have some rule that enforces space in parens for if conditionals only and nothing else.

**Comment by not-an-aardvark** [comment]
Thanks for creating an issue and for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
